internal/config/template.go

```diff
@@ -0,0 +1,256 @@
+package config
+
+import (
+	"gopkg.in/yaml.v2"
+)
+
+// Structs generated from template/imagehub-filter.yaml
+// EnvoyFilter
+type EnvoyFilter struct {
```

Comment:
@DelusionalOptimist perhaps use this - https://pkg.go.dev/istio.io/client-go@v1.9.4/pkg/apis/networking/v1alpha3#EnvoyFilter

Reply:
Yes @kumarabd, probably we should use this here. In fact, I tried experimenting with using protobufs for this recently but being fairly new to them, I got stuck.

Tying to parse the yaml to protobufs provided by istio and envoy - https://codeberg.org/DelusionalOptimist/envoy-filter-gen.
I think the next step in this would be parsing the obtained "value" stuct to [TypedExtensionConfig](https://github.com/envoyproxy/go-control-plane/blob/main/envoy/config/core/v3/extension.pb.go#L33) implemented in envoy's go-control-plane but I'm clueless about how this should be done. Any suggestions?

Reply:
Can you explain me the use case? perhaps a call sounds good. @DelusionalOptimist

Reply:
Sure @kumarabd! Whenever you say :+1:

Reply:
Gents, how did we make out here?

Reply:
Yes, we had a positive discussion about using protobufs. Was a little occupied for the last few days but that's done now so I'll be resuming work on this. Shouldn't be much work left :slightly_smiling_face:

Reply:
Oh, very good. Ok. Signal when ready for a review.